### ISSUE TITLE
TB addressbook: The "plain-text" option is set on nearly all contacts

### ISSUE THREAD

**Issue by mMuck** [jobisoft/DAV-4-TbSync#43]
## Your environment

TbSync version: 0.7.22
DAV-4-TbSync version: 0.10
Thunderbird version: 60.3.1 (32-bit)

[ ] Yes, I have installed the latest available (beta) version from 
 - https://github.com/jobisoft/TbSync/releases 
 - https://github.com/jobisoft/DAV-4-TbSync/releases 
and my issue is not yet fixed, I can still reproduce it.

[X] Ahmm, Thunderbird show the same version numbers installed as the versions stated at the releases list. So, is it true that I have the latest releases?

## Expected behavior
I connected an addressbook of a Nextcloud installation to TB. I use this Add-On as a replacement for the SoGo Connector. The imported contact should be the same as with the old interface :wink:.

## Actual behavior
The "text-only" option is set on nearly all contacts but should be set to "unknown" (or where does the information come from? Does CardDAV hold this? The consequences are very bad: Over the last three weeks I had many problems with missing images distorted HTML signatures and I didn't found out the reason for missing HTML in sent mails composed in a HTML window. Took a bit of time to figure out...

## Steps to reproduce
Sync to a CardDAV addressbook and check the "text-only" option at the imported contacts.

If you need the log, please ask for it. Thanks.

**Comment by jobisoft** [comment]
Did not know about that text-only option. Where can the user set/see that value? Screenshot?

**Comment by jobisoft** [comment]
Is there a corresponding vCard v3 field to sync this information from/to?

**Comment by mMuck** [comment]
Can't answer this. Sorry. I never saw this information on Nextcloud but it shows only a subset of the vCard fields too...

**Comment by brad2014** [comment]
I believe this refers to the X-MOZILLA-HTML:FALSE entry in a vcard3, which it appears is syncing properly.

**Comment by jobisoft** [comment]
Oha, I DO Sync that field:
https://github.com/jobisoft/DAV-4-TbSync/blob/master/content/tools.js#L673

@mMuck Can you confirm, that changes to this value in TB are properly synced? sync, unsubscribe the book in question, resubscribe and check the values? 

This would also mean, that your cards on the server had the text-only property stored.

**Comment by mMuck** [comment]
But the same data in the Nextcloud server was previously synced with SoGo Connector an now with TbSync. I never set any information for a preferred mail format. I will look at the actual involved vCard raw data...

**Comment by jobisoft** [comment]
For reference
http://doxygen.db48x.net/comm-central/html/interfacensIAbPreferMailFormat.html

**Comment by jobisoft** [comment]
And this is how I handle it:
https://github.com/jobisoft/DAV-4-TbSync/blob/master/content/tools.js#L917

**Comment by brad2014** [comment]
I've confirmed that a nextcloud 14 contact (created via the nextcloud interface) has no X-MOZILLA-HTML entry, and when synced, Thunderbird shows the address card html preference as "unknown".

@mmuck, In Thunderbird, you may want to set the checkbox on  Settings->Composition->General->Send options...->Send as plain text when possible (Nachrichten falls moeglich als Reintext senden), 
and set "Send the message as both plaintext and html".

In that way, everyone gets both formats.  Note that on Thunderbird, if you receive an email containing both plaintext and html, it will be displayed as one or the other depending on your choice of View->Message Body As  set to "Original HTML" or "Plain Text".  It can be disconcerting if you accidentally leave this set to "Plain Text".  All the HTML apparently disappears.

**Comment by mMuck** [comment]
> I've confirmed that a nextcloud 14 contact (created via the nextcloud interface) has no X-MOZILLA-HTML entry, and when synced, Thunderbird shows the address card html preference as "unknown".

I can confirm this too!

> @mMuck, In Thunderbird, you may want to set the checkbox on Settings->Composition->General->Send options...->Send as plain text when possible (Nachrichten falls moeglich als Reintext senden),
> and set "Send the message as both plaintext and html".

I switched this off because I assumed the error in this automated decision...

I investigated this earlier in the morning. I looked in the vCard formatted entries in the Nextcloud database. I've no idea where the "X-MOZILLA-HTML: FALSE" entries came from. I think this is a SoGo connector bug (https://github.com/inverse-inc/sogo-connector.tb31) and has nothing to do with TbSync :smiley:. Good for you :+1: The handling of this preference seems to be correct.
I batch edited this "plain-text" tag in the database (because this affected nearly all entries, I think all with a mail address), deleted the TbSync Account, reconnected it and: Everything is fine.
For me, this issue could be closed. Unfortunately I can't recognize the reason or time when the "X-MOZILLA-HTML: FALSE" entries were added in the database.
